-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4eaafe/16275/summary.html
**COMMIT**: 4fe7330eae9a82ef16c33ed1871393bd53f3d264
**CMSSW**: CMSSW_12_0_X_2021-06-28-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/34270/16275/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-06-28-1100/src/Alignment/CommonAlignmentProducer/src/AlignmentSeedSelector.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-06-28-1100/src/Alignment/CommonAlignmentProducer/src/AlignmentTrackSelector.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-06-28-1100/src/Alignment/CommonAlignmentProducer/src/AlignmentTracksFromVertexSelector.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-06-28-1100/src/Alignment/CommonAlignmentProducer/src/AlignmentTwoBoyDecayTrackSelector.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-06-28-1100/src/Alignment/CommonAlignmentProducer/src/AlignmentProducerBase.cc: In member function 'void AlignmentProducerBase::createMonitors()':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-06-28-1100/src/Alignment/CommonAlignmentProducer/src/AlignmentProducerBase.cc:295:106: error: no matching function for call to 'edmplugin::PluginFactory<AlignmentMonitorBase*(const edm::ParameterSet&, edm::ConsumesCollector&)>::create(const std::__cxx11::basic_string<char>&, const edm::ParameterSet&)'
  295 |         AlignmentMonitorPluginFactory::get()->create(miter, monitorConfig.getUntrackedParameterSet(miter)));
      |                                                                                                          ^
In file included from /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc900/cms/cmssw-patch/CMSSW_12_0_X_2021-06-28-1100/src/Alignment/CommonAlignmentAlgorithm/interface/AlignmentAlgorithmPluginFactory.h:10,
                 from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-06-28-1100/src/Alignment/CommonAlignmentProducer/src/AlignmentProducerBase.cc:5:
/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc900/cms/cmssw-patch/CMSSW_12_0_X_2021-06-28-1100/src/FWCore/PluginManager/interface/PluginFactory.h:63:24: note: candidate: 'std::unique_ptr<_Tp> edmplugin::PluginFactory<R*(Args ...)>::create(const string&, Args ...) const [with R = AlignmentMonitorBase; Args = {const edm::ParameterSet&, edm::ConsumesCollector&}; std::string = std::__cxx11::basic_string<char>]'
</pre>

--- a/Alignment/CommonAlignmentProducer/interface/AlignmentProducerBase.h
+++ b/Alignment/CommonAlignmentProducer/interface/AlignmentProducerBase.h
@@ -138,7 +138,7 @@ private:
   void createAlignmentAlgorithm(edm::ConsumesCollector&);
 
   /// Creates the monitors
-  void createMonitors();
+  void createMonitors(edm::ConsumesCollector&);
 
   /// Creates the calibrations
   void createCalibrations();

--- a/Alignment/CommonAlignmentProducer/src/AlignmentProducerBase.cc
+++ b/Alignment/CommonAlignmentProducer/src/AlignmentProducerBase.cc
@@ -92,7 +92,7 @@ AlignmentProducerBase::AlignmentProducerBase(const edm::ParameterSet& config, ed
   }
 
   createAlignmentAlgorithm(iC);
-  createMonitors();
+  createMonitors(iC);
   createCalibrations();
 }
 
@@ -287,12 +287,12 @@ void AlignmentProducerBase::createAlignmentAlgorithm(edm::ConsumesCollector& iC)
 }
 
 //------------------------------------------------------------------------------
-void AlignmentProducerBase::createMonitors() {
+void AlignmentProducerBase::createMonitors(edm::ConsumesCollector& iC) {
   const auto& monitorConfig = config_.getParameter<edm::ParameterSet>("monitorConfig");
   auto monitors = monitorConfig.getUntrackedParameter<std::vector<std::string> >("monitors");
   for (const auto& miter : monitors) {
     monitors_.emplace_back(
-        AlignmentMonitorPluginFactory::get()->create(miter, monitorConfig.getUntrackedParameterSet(miter)));
+        AlignmentMonitorPluginFactory::get()->create(miter, monitorConfig.getUntrackedParameterSet(miter), iC));
   }
 }
 

